app/services/user/user_service.py

```diff
@@ -1,5 +1,5 @@
 from sqlalchemy.orm import Session
-from app.models import UserModel, AutoPasswordAuthModel
+from app.models import UserModel, AutoPasswordAuthModel, OnyenPIDModel
```

Comment:
Onyen and pid should be unassociated from user during deletion.

Reply:
I am doing this within the student/instructor sync methods already. 

It belongs more in LMS Sync - the whole reason we are associating/associating pids is because Canvas returns a pid. It's LMS specific logic

Reply:
That's a valid point, I would prefer it in LMS sync. The worry here is that user deletion outside of LMS sync will no longer work, since the onyen pid model has a foreign key to the user's onyen. Thus, it will error if you try to delete the user before deleting the onyen-pid association (i.e. through the user service, rather than the LMS service).

Reply:
For now, you should add it to user deletion as well (duplicate the changes). Make sure it's in a try/catch since it will throw if the associated was already deleted (as it will be). This will be superseded by the cascade delete ticket, which will handle all of this automatically.